File: tests/python_service/src/python_service/minimal_service.py
from .basic_gen.minimal_interface import MinimalInterface
from .basic_gen.backend_minimal_client import BackendMinimalClient

def get_minimal_service_and_backend_client():
    service = MinimalInterface()
    backend_client = BackendMinimalClient()
    async def bump_handler():
        return await backend_client.Bump()
    def on_bumped():
        service.Bumped()
        print("Bump emitted")
    service.on_Bump(bump_handler)
    backend_client.on_Bumped(on_bumped)
    return service, backend_client

File: tests/csharp_service/MinimalService.cs
namespace csharp_service;
using TestService.Generated;

public class MinimalServiceWorker : BackgroundService
{
    private readonly ILogger<MinimalServiceWorker> _logger;

    public MinimalServiceWorker(ILogger<MinimalServiceWorker> logger)
    {
        _logger = logger;
    }

    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
    {
        var client = new BackendMinimalClient();

        var service = new MinimalInterface();
        service.OnBump = async ()=>{
            await client.BumpAsync();
        };

        client.Bumped += ()=>{service.EmitBumped();};
        var connection = new Connection();
        await connection.RegisterInterfaceAsync(service);
        await connection.RegisterClient(client);
        await connection.ConnectAsync(stoppingToken);
    }
}